### PR TITLE
fix: account linking with 2FA

### PR DESCRIPTION
**Pull request by hperl**
This fixes some edge cases with OIDC account linking for accounts with 2FA enabled.

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: selfservice/strategy/oidc/strategy_settings.go [hunk 1]
```diff
@@ -550,3 +551,47 @@ func (s *Strategy) Link(ctx context.Context, i *identity.Identity, credentialsCo
...
 	return nil
 }
+
+func (s *Strategy) CompletedLogin(sess *session.Session, data *flow.DuplicateCredentialsData) error {
+	var credentialsOIDCConfig identity.CredentialsOIDC
+	if err := json.Unmarshal(data.CredentialsConfig, &credentialsOIDCConfig); err != nil {
+		return err
+	}
+	if len(credentialsOIDCConfig.Providers) != 1 {
+		return errors.New("no oidc provider was set")
```
Comment (aeneasr) [inline selfservice/strategy/oidc/strategy_settings.go:561]: What if this is 2?

**Review by aeneasr — COMMENTED** [review]
Just two minor things otherwise LGTM

File: selfservice/strategy/oidc/strategy_settings.go (hunk 1 above)
Comment (hperl) [reply selfservice/strategy/oidc/strategy_settings.go:561]: it is an invalid state then, we ensure it is always 1.

Comment (hperl) [reply selfservice/strategy/oidc/strategy_settings.go:561]: here: https://github.com/ory/kratos/blob/be5277680908f33f5a6db09e6e88e4dbbc7aeab9/selfservice/strategy/oidc/strategy_settings.go#L584

**Review by aeneasr — APPROVED** [review]

**Pull request merged**